src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -114,6 +114,12 @@ public static PackageSpec GetPackageSpec(Stream stream, string name, string pack
             packageSpec.Summary = rawPackageSpec.GetValue<string>("summary");
             packageSpec.ReleaseNotes = rawPackageSpec.GetValue<string>("releaseNotes");
 
+            var buildOptions = rawPackageSpec["buildOptions"] as JObject;
+            if (buildOptions != null)
+            {
+                packageSpec.OutputName = buildOptions.GetValue<string>("outputName");
```

Comment:
This needs a positive and negative unit test (just on `JsonPackageSpecReader`). Also, what happens when it's not a  string? An object? Non-existent? JSON `null`?